app/api/entries/[id]/route.js

```diff
@@ -0,0 +1,40 @@
+import { dbConnect } from '@/utils/dbConnects';
+import Entry from '../models/Entry';
+import { getUserFromCookie } from '@/utils/auth';
+export async function DELETE(req) {
+  await dbConnect();
+  const user = getUserFromCookie();
+  if (!user) {
+    return new Response(JSON.stringify({ error: 'Unauthorized' }), {
+      status: 401,
+    });
+  }
+  const url = new URL(req.url);
+  const entryId = url.pathname.split('/').pop();
```

Comment:
This is not the recommended way by next js.
https://nextjs.org/blog/building-apis-with-nextjs#5-dynamic-routes
You should instead use what's provided by the package 
```
  const entryId = (await params).id;
```